### PR TITLE
Improve non strict in_array

### PR DESCRIPTION
**Pull request by VincentLanglet**
Closes https://github.com/phpstan/phpstan/issues/9436

### CODE REVIEW

File: tests/PHPStan/Analyser/nsrt/in_array_loose.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		float $float,
+		bool $bool,
+		string|int $stringOrInt,
+		string|null $stringOrNull,
+	): void {
+		if (in_array($string, ['1', 'a'])) {
+			assertType("'1'|'a'", $string);
+		}
+		if (in_array($string, [1, 'a'])) {
+			assertType("string", $string);
```
Comment (VincentLanglet) [inline tests/PHPStan/Analyser/nsrt/in_array_loose.php]: Some assertion will be improved later for example when https://phpstan.org/r/49e37d0b-b17a-4a90-9fff-0618a5c7fc41 will be improved.

Comment (VincentLanglet) [inline tests/PHPStan/Analyser/nsrt/in_array_loose.php]: Some assertion will be improved later for example when https://phpstan.org/r/49e37d0b-b17a-4a90-9fff-0618a5c7fc41 will be improved.

Comment (staabm) [reply tests/PHPStan/Analyser/nsrt/in_array_loose.php]: Usually I would drop a comment in the test code like `could be xyz` to make people aware that a more precise result is expected.

The review comment itself is usually not that helpful

**Comment by ondrejmirtes** [comment]
Fixes https://github.com/phpstan/phpstan/issues/9752 too, please add a regression test.

**Comment by VincentLanglet** [comment]
> Fixes [phpstan/phpstan#9752](https://github.com/phpstan/phpstan/issues/9752) too, please add a regression test.

Hi, 

It fixes this comment https://github.com/phpstan/phpstan/issues/9752#issuecomment-1697288324
with the snippet https://phpstan.org/r/4cbdb7cd-ab24-4a52-a816-b1e378cc11b4

but it doesn't fix the original issue https://github.com/phpstan/phpstan/issues/9752#issue-1847075031 with the snippet https://phpstan.org/r/341e08ac-54fe-4f45-9586-703c1707060c if I understand correctly the issue bot.

So it doesn't close https://github.com/phpstan/phpstan/issues/9752.
Do you still want a non-regression test for the snippet it fixes ?

**Comment by ondrejmirtes** [comment]
Sorry I didn't notice it, forget what I said, we don't need a test for that :)

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**